CMakeLists.txt

```diff
@@ -34,6 +34,8 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_RELEASE "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}")
 string(TOLOWER "${CMAKE_BUILD_TYPE}" cmake_build_type_lower)
 
 if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+  add_definitions(-DMULTIPASS_COMPILER_CLANG)
```

Comment:
What about AppleClang?

Also, I would rather see this closer to where the `MULTIPASS_PLATFORM_...` vars are defined, to be sure they are coherent. So in this [if-else](https://github.com/canonical/multipass/blob/a11c10192a0192ec998431672a2975619addcf5b/CMakeLists.txt#L156). WDYT?

Reply:
Well, both `MULTIPASS_COMPILER_CLANG` and `MULTIPASS_PLATFORM_MACOS` defined will be equivalent to `MULTIPASS_COMPILER_APPLE_CLANG`. I my point of view, I'll add that flag if Clang and Apple Clang differ substantially, making important to have such flag. Do you think it's worth the effort of adding such a flag?

OTOH, answering to your question, maybe detecting compiler as part of the platform would simplify a bit the code. Let me try to do it.

Reply:
`MULTIPASS_COMPILER_CLANG` + `MULTIPASS_PLATFORM_MACOS` is fine by me, but I am not sure that's what would be defined with that code? IOW, does `CMAKE_CXX_COMPILER_ID STREQUAL "Clang"` with AppleClang? Otherwise we'd say MSVC

Reply:
Oh, good question. The answer is no, see [here](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_ID.html).

Reply:
Fixed.

Reply:
Fixed